src/components/Todolist.tsx

```diff
@@ -0,0 +1,55 @@
+import { Filter } from '../types/Filter';
+import { Todo } from '../types/Todo';
+
+interface TodoListProps {
+  todos: Todo[],
+  filter: Filter,
+}
+
+export const TodoList: React.FC<TodoListProps> = ({ todos, filter }) => {
+  const filteredTodos = todos.filter(todo => {
```

Comment:
Same about the useMemo(), use it here. 
But it's important to note that useMemo is useful when the calculation is time-consuming or resource-intensive. If the calculation is relatively simple and fast, using useMemo may not provide significant benefits.